In [2]:
import pandas as pd

df = pd.read_csv('retail_sales_dataset.csv')
df.head()

,Transaction ID,Date,Customer ID,Gender,Age,Product Category,Quantity,Price per Unit,Total Amount
0,1,2023-11-24,CUST001,Male,34,Beauty,3,50,150
1,2,2023-02-27,CUST002,Female,26,Clothing,2,500,1000
2,3,2023-01-13,CUST003,Male,50,Electronics,1,30,30
3,4,2023-05-21,CUST004,Male,37,Clothing,1,500,500
4,5,2023-05-06,CUST005,Male,30,Beauty,2,50,100


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Transaction ID    1000 non-null   int64 
 1   Date              1000 non-null   object
 2   Customer ID       1000 non-null   object
 3   Gender            1000 non-null   object
 4   Age               1000 non-null   int64 
 5   Product Category  1000 non-null   object
 6   Quantity          1000 non-null   int64 
 7   Price per Unit    1000 non-null   int64 
 8   Total Amount      1000 non-null   int64 
dtypes: int64(5), object(4)
memory usage: 70.4+ KB


In [4]:
df.describe()

,Transaction ID,Age,Quantity,Price per Unit,Total Amount
count,1000.000000,1000.00000,1000.000000,1000.000000,1000.000000
mean,500.500000,41.39200,2.514000,179.890000,456.000000
std,288.819436,13.68143,1.132734,189.681356,559.997632
min,1.000000,18.00000,1.000000,25.000000,25.000000
25%,250.750000,29.00000,1.000000,30.000000,60.000000
50%,500.500000,42.00000,3.000000,50.000000,135.000000
75%,750.250000,53.00000,4.000000,300.000000,900.000000
max,1000.000000,64.00000,4.000000,500.000000,2000.000000


In [5]:
df['Gender'].value_counts()

Gender
Female    510
Male      490
Name: count, dtype: int64

In [6]:
df['Product Category'].value_counts()

Product Category
Clothing       351
Electronics    342
Beauty         307
Name: count, dtype: int64

In [7]:
df.groupby('Product Category')['Total Amount'].sum()

Product Category
Beauty         143515
Clothing       155580
Electronics    156905
Name: Total Amount, dtype: int64

In [8]:
df.groupby('Product Category')['Price per Unit'].mean()

Product Category
Beauty         184.055375
Clothing       174.287749
Electronics    181.900585
Name: Price per Unit, dtype: float64

In [9]:
df.groupby('Product Category')['Quantity'].mean()

Product Category
Beauty         2.511401
Clothing       2.547009
Electronics    2.482456
Name: Quantity, dtype: float64

In [10]:
df['Age Group'] = pd.cut(df['Age'], bins=[17, 25, 35, 45, 55, 65], labels=['18-25', '26-35', '36-45', '46-55', '56-65'])
df.groupby('Age Group')['Total Amount'].sum()

C:\Users\surys\AppData\Local\Temp\ipykernel_18820\2284543156.py:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby('Age Group')['Total Amount'].sum()


Age Group
18-25     84550
26-35     98480
36-45     91870
46-55    100690
56-65     80410
Name: Total Amount, dtype: int64

In [11]:
df.groupby('Age Group', observed=False)['Total Amount'].sum()

Age Group
18-25     84550
26-35     98480
36-45     91870
46-55    100690
56-65     80410
Name: Total Amount, dtype: int64

## EDA Findings — Retail Sales Dataset

**Dataset Overview:**
- 1000 transactions, 9 columns, no missing values
- Categories: Clothing, Electronics, Beauty (fairly balanced)
- Gender split: 510 Female, 490 Male (balanced)

**Key Insights:**

1. **Electronics generates the highest revenue** (₹156,905) despite having fewer transactions than Clothing, and having similar average price-per-unit and quantity-per-transaction to other categories — suggesting revenue is driven by a combination of small factors rather than one dominant variable.

2. **Price-per-unit and quantity purchased are consistent across all product categories** (~₹175-184 avg price, ~2.5 avg quantity), showing customer purchasing behavior doesn't vary much by category.

3. **Age 46-55 is the top-spending demographic** (₹100,690 total), followed by 26-35 (₹98,480). Younger (18-25) and older (56-65) customers spend the least — consistent with peak-earning-age spending patterns.

4. **Total Amount has high variability** (std ₹560 vs mean ₹456), meaning spending per transaction is quite spread out rather than clustered near the average — some customers make small purchases, others large ones.

**Conclusion:** No single factor (price, quantity, gender) strongly drives category-level revenue differences — Electronics' lead appears to come from a combination of small advantages. Age is the clearest differentiator in customer spending behavior.